pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/withscript/WithScriptDescribable.java

```diff
@@ -55,20 +52,7 @@ public WithScriptScript getScript(CpsScript cpsScript) throws Exception {
         if (c == null)
             throw new IllegalStateException("Expected to be called from CpsThread");
 
-        Class clz;
-        try {
-            clz = cpsScript.getClass().getClassLoader().loadClass(getDescriptor().getScriptClass());
-        } catch (ClassNotFoundException e) {
-            // This is special casing to deal with PluginFirstClassLoaders, which don't have a functional findResource method.
-            // That results in GroovyClassLoader.loadClass failing to find resources to parse and load.
-            // TODO delete JENKINS-44898 workaround as of Jenkins 2.66
-            URL res = getDescriptor().getClass().getClassLoader().getResource(getDescriptor().getScriptClass().replace('.', '/') + ".groovy");
-            if (res != null) {
-                clz = ((GroovyClassLoader) cpsScript.getClass().getClassLoader()).parseClass(new GroovyCodeSource(res));
-            } else {
-                throw e;
-            }
```

Comment:
Strictly speaking, we don't have to remove this fallback. However, it is obsolete as far as I can tell, and the fallback interacts weirdly with SECURITY-359, since it causes any files that are rejected by the allowlist to be loaded using a regular Java class loader and then parsed it using the _untrusted_ `GroovyClassLoader` for the Pipeline, causing the Groovy source file to be sandbox-transformed, which leads to unexpected `RejectedAccessException`s and confusing error messages.